pkg/config/redis.go

```diff
@@ -41,10 +48,21 @@ func (r *Redis) NewClient(logger *zap.SugaredLogger) (*icingaredis.Client, error
 }
 
 // dialWithLogging returns a Redis Dialer with logging capabilities.
-func dialWithLogging(logger *zap.SugaredLogger) func(context.Context, string, string) (net.Conn, error) {
+func dialWithLogging(
+	tlsConfig *tls.Config, logger *zap.SugaredLogger,
+) func(context.Context, string, string) (net.Conn, error) {
+	var dl interface {
+		DialContext(ctx context.Context, network, address string) (net.Conn, error)
+	}
+
+	if tlsConfig == nil {
+		dl = &net.Dialer{}
+	} else {
+		dl = &tls.Dialer{Config: tlsConfig}
+	}
```

Comment:
`redis.Options` has a member `TLSConfig`, that should probably be used instead: https://pkg.go.dev/github.com/go-redis/redis/v8#Options